File: Day5/find_lucky_integer_in_an_array.cc
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findLucky(vector<int>& arr) {
        unordered_map<int, int> freq;
        for (int num : arr) {
            freq[num]++;
        }

        int luckyNum = -1;
        for (auto& [key, val] : freq) {
            if (key == val) {
                luckyNum = max(luckyNum, key);
            }
        }
        return luckyNum;
    }
};

File: Day5/find_lucky_integer_in_an_array.java
import java.util.*;

class Solution {
    public int findLucky(int[] arr) {
        Map<Integer, Integer> freq = new HashMap<>();
        for (int num : arr) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }

        int luckyNum = -1;
        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            if (entry.getKey().equals(entry.getValue())) {
                luckyNum = Math.max(luckyNum, entry.getKey());
            }
        }
        return luckyNum;
    }
}
